File: api/ti84c.cxx
/**
 *  Java Grinder
 *  Author: Michael Kohn
 *   Email: mike@mikekohn.net
 *     Web: http://www.mikekohn.net/
 * License: GPL
 *
 * Copyright 2014 by Michael Kohn
 *
 */

#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <stdint.h>

#include "JavaClass.h"
#include "cpu.h"

#define CHECK_FUNC(funct) \
  if (strncmp(#funct, function, sizeof(#funct)-1) == 0) \
  { \
    return ti84c_##funct(java_class, generator); \
  }

#if 0
static int ti84c_setClock16(JavaClass *java_class, Generator *generator)
{
  return generator->ti84c_setClock16();
}
#endif

static int ti84c_clearRect(JavaClass *java_class, Generator *generator)
{
  return generator->ti84c_clearRect();
}

static int ti84c_drawHL(JavaClass *java_class, Generator *generator)
{
  return generator->ti84c_drawHL();
}

static int ti84c_drawLine(JavaClass *java_class, Generator *generator)
{
  return generator->ti84c_drawLine();
}

static int ti84c_drawPoint(JavaClass *java_class, Generator *generator)
{
  return generator->ti84c_drawPoint();
}

static int ti84c_drawString(JavaClass *java_class, Generator *generator)
{
  return generator->ti84c_drawString();
}

static int ti84c_drawStringCenter(JavaClass *java_class, Generator *generator)
{
  return generator->ti84c_drawStringCenter();
}

static int ti84c_fillRect(JavaClass *java_class, Generator *generator)
{
  return generator->ti84c_fillRect();
}

static int ti84c_setCursorX(JavaClass *java_class, Generator *generator)
{
  return generator->ti84c_setCursorX();
}

static int ti84c_setCursorY(JavaClass *java_class, Generator *generator)
{
  return generator->ti84c_setCursorY();
}

static int ti84c_setDrawBGColor(JavaClass *java_class, Generator *generator)
{
  return generator->ti84c_setDrawBGColor();
}

static int ti84c_setDrawBGWhite(JavaClass *java_class, Generator *generator)
{
  return generator->ti84c_setDrawBGWhite();
}

static int ti84c_setDrawColor(JavaClass *java_class, Generator *generator)
{
  return generator->ti84c_setDrawColor();
}

static int ti84c_setTextBGColor(JavaClass *java_class, Generator *generator)
{
  return generator->ti84c_setTextBGColor();
}

static int ti84c_setTextColor(JavaClass *java_class, Generator *generator)
{
  return generator->ti84c_setTextColor();
}

int ti84c(JavaClass *java_class, Generator *generator, char *function)
{
  CHECK_FUNC(clearRect)
  CHECK_FUNC(drawHL)
  CHECK_FUNC(drawLine)
  CHECK_FUNC(drawPoint)
  CHECK_FUNC(drawString)
  CHECK_FUNC(drawStringCenter)
  CHECK_FUNC(fillRect)
  CHECK_FUNC(setCursorX)
  CHECK_FUNC(setCursorY)
  CHECK_FUNC(setDrawBGColor)
  CHECK_FUNC(setDrawBGWhite)
  CHECK_FUNC(setDrawColor)
  CHECK_FUNC(setTextBGColor)
  CHECK_FUNC(setTextColor)

  return -1;
}

File: java/net/mikekohn/java_grinder/TI84C.java
/**
 *  Java Grinder
 *  Author: Michael Kohn
 *   Email: mike@mikekohn.net
 *     Web: http://www.naken.cc/
 * License: GPL
 *
 * Copyright 2014 by Michael Kohn
 *
 */

package net.mikekohn.java_grinder;

// http://wikiti.brandonw.net/index.php?title=84PCSE:OS:Applications

abstract public class TI84C
{
  protected TI84C()
  {
  }

  public static void clearRect(int x0, int x1, int y0, int y1) { }
  public static void drawHL() { }
  public static void drawLine(int x0, int y0, int x1, int y1, int operation) { }
  public static void drawPoint(int x, int y, int operation) { }
  public static void drawString(String text) { }
  public static void drawStringCenter(String text) { }
  public static void fillRect(int x0, int x1, int y0, int y1) { }
  public static void setCursorX(int x) { }
  public static void setCursorY(int y) { }
  public static void setDrawBGColor(int color) { }
  public static void setDrawBGWhite() { }
  public static void setDrawColor(int color) { }
  public static void setTextBGColor(int color) { }
  public static void setTextColor(int color) { }

  public static final int OPERATION_ERASE_PIXEL = 0;
  public static final int OPERATION_DRAW_PIXEL = 1;
  public static final int OPERATION_INVERT_PIXEL = 2;
  public static final int OPERATION_TEST_PIXEL = 3;
  public static final int OPERATION_RESTORE_PIXEL = 4;
}

